src/Drivers/nlpPriDec_ex9.cpp

```diff
@@ -0,0 +1,198 @@
+
+  nlp.options->SetIntegerValue("verbosity_level", 1);
+  nlp.options->SetNumericValue("mu0", 1e-1);
+  //nlp.options->SetNumericValue("tolerance", 1e-5);
+  
+  hiopAlgFilterIPMNewton solver(&nlp);
+
+  status = solver.run();
+
+  obj_ = solver.getObjective();
+  solver.getSolution(x);
+
+  if(status<0) {
+    printf("solver returned negative solve status: %d (with objective is %18.12e)\n", status, solver.getObjective());
+    return status;
+  }
+  
+  // print x
+  // for(int i=0;i<nx_;i++) printf(" %d %18.12e ",i,x[i]);
+    
+  if(sol_==NULL) {
+    sol_ = new double[nx_];
+  }
+  memcpy(sol_,x, nx_*sizeof(double));
+  
+  //compute the recourse estimate
+  if(include_r) {
+    double rec_appx = 0.;
+    basecase_->get_rec_obj(nx_, x, rec_appx);
+    //printf("recourse estimate: is %18.12e\n", rec_appx); 
+  }
+  
+  return Solve_Success;
+  //return hiop::SolverInternal_Error;
+}
+
+bool PriDecMasterProblemEx9::
+set_recourse_approx_evaluator(const int n, 
+		              hiopInterfacePriDecProblem::RecourseApproxEvaluator* evaluator)
+{
+  assert(n==nc_);
+  basecase_->set_quadratic_terms(n, evaluator);
+  return true; 
+}
+
+bool PriDecMasterProblemEx9::eval_f_rterm(size_t idx, const int& n, const double* x, double& rval)
+{
+  assert(nx_==n);
+  rval=-1e+20;
+  hiopSolveStatus status;
+  double* xi;
+  
+#ifdef HIOP_USE_MPI
+  double t3 =  MPI_Wtime(); 
+  double t4 = 0.; 
+#endif 
+  
+  // xi can be set below 
+  xi = new double[nS_]; 
+  for(int i=0;i<nS_;i++) xi[i] = 1.;
+  //xi[ny_-1] = double(idx/100.0+1.);
+
+  PriDecRecourseProblemEx9* ex9_recourse;
+
+  ex9_recourse = new PriDecRecourseProblemEx9(nc_, nS_,S_,x,xi);
+  
+  // set a few contingencies to have different sparse structure
+  /*
+  if(idx%30==0) {  
+    ex9_recourse->set_sparse(0.3);
+  }
+  */
+  
+  hiopNlpMDS nlp(*ex9_recourse);
+  nlp.options->SetStringValue("duals_update_type", "linear");
+  //nlp.options->SetStringValue("dualsInitialization", "zero");
+  nlp.options->SetStringValue("Hessian", "analytical_exact");
+#ifdef HIOP_USE_GPU
+  nlp.options->SetStringValue("compute_mode", "hybrid");
+#else  
+  nlp.options->SetStringValue("compute_mode", "cpu");
+#endif
+  //nlp.options->SetStringValue("time_kkt", "on");
+  nlp.options->SetIntegerValue("verbosity_level", 1);
+  nlp.options->SetNumericValue("mu0", 1e-1);
+  //nlp.options->SetNumericValue("tolerance", 1e-5);
+
+  hiopAlgFilterIPMNewton solver(&nlp);
+
+  //assert("for debugging" && false); //for debugging purpose
+  status = solver.run();
+  rval = solver.getObjective();  
+  if(y_==NULL) {
+    y_ = new double[ny_];
+  }
+  solver.getSolution(y_);
+  
+  #ifdef HIOP_USE_MPI
+  // uncomment if want to monitor contingency computing time
+  /* t4 =  MPI_Wtime(); 
+     if(idx==0||idx==1) {
+       printf( "Elapsed time for contingency %d is %f\n",idx, t4 - t3 ); 
+       printf(" Objective for idx %d value %18.12e, xi %18.12e\n",idx,rval,xi[0]);
+     }
+  */
```

Comment:
Perhaps better solution would be to add a command line option to turn on contingency compute time monitoring.